### ISSUE TITLE
YOLO version 3 equivalent in Keras 

### ISSUE THREAD

**Issue by hiba007** [experiencor/basic-yolo-keras#207]
Can anybody help me in building YOLOv3 architecture in keras?
here is the link
I think its residual network
https://github.com/pjreddie/darknet/blob/master/cfg/yolov3.cfg
@experiencor

**Comment by experiencor** [comment]
@hiba007 ResNet backend is currently supported. Please check the readme. If possible, can you provide a drawing of YOLOv3.

**Comment by txdat** [comment]
after reading, i found some main different things btw 2 and 3:
- new backbone network (similar to resnet)
- using feature pyramid as main concept
- using anchors' size from input size, not output size (yolo2) -> new anchors 's size = 32x
- at each feature map of pyramid, use 3 different anchors (different scales) -> total 9 anchors on 3 levels of pyramid -> reduce computation, instead of 5 or 9 anchors at every level

**Comment by hiba007** [comment]
drawing means architecture ?
here is cfg file 
https://github.com/pjreddie/darknet/blob/master/cfg/yolov3.cfg

**Comment by hiba007** [comment]
you mean this? @experiencor 
![yolov3 1](https://user-images.githubusercontent.com/37494476/37933054-f0a38e00-30fe-11e8-8bf7-1da46ad916f3.JPG)
![yolov3 2](https://user-images.githubusercontent.com/37494476/37933074-03248b60-30ff-11e8-8777-1949751e4be4.JPG)
![yolov3 3](https://user-images.githubusercontent.com/37494476/37933085-0f7dcf16-30ff-11e8-94b8-f77b05296a57.JPG)
![yolov3 4](https://user-images.githubusercontent.com/37494476/37933098-1840cc02-30ff-11e8-922a-e853b16b97ec.JPG)

**Comment by experiencor** [comment]
@hiba007 @txdat Interesting changes! I'll take a closer look.

**Comment by cthacker** [comment]
Yolov3 also doesn't use softmax and just uses logistic units for each class (multiclass classification)

**Comment by zeynali** [comment]
what's this mean : 
`using anchors' size from input size, not output size (yolo2) -> new anchors 's size = 32x`

**Comment by experiencor** [comment]
in version 3, anchor sizes are absolute, not relative to grid size like in version 1 and 2

**Comment by zeynali** [comment]
what are the advantage of this method rather relative?

**Comment by experiencor** [comment]
@zeynali I'm not very clear about this. Still working on the new loss function.

**Comment by experiencor** [comment]
@hiba007 @txdat YOLOv3 detection is out. Still working on the new multi-scale loss training.

**Comment by zeynali** [comment]
@experiencor , Thanks a a lot , When will the multi-scale training be prepared?
please explain about differences between v3 and v2 ? more details if possible. Thanks

**Comment by cthacker** [comment]
@experiencor New model looks good.

 why do you resize the image but keep the aspect ratio the same and embed into a letterbox? Do you find that helps with accuracy?

Also are you planning to implement the lossfn for this network or no? I'm still trying to figure out everything that is different compared the standard yolo loss.

**Comment by zeynali** [comment]
@cthacker , Is not loss function corresponding to Version 3?

**Comment by txdat** [comment]
@experiencor 2 things i dont understand in yolo3_detect.py:
1) netout[..., 5:]  = netout[..., 4][..., np.newaxis] * netout[..., 5:]'
why do you multiply iou_score to class_score?
2) you compute overlaps in (0. 1) scale not input's scale?

**Comment by experiencor** [comment]
1. It's P(Class) = P(Class|Object) * P(Object)
2. When you compute IOU, it does not matter.
3. I don't why warmup is omitted in the new loss function. Asked a question but not yet get the answer.

**Comment by experiencor** [comment]
https://github.com/experiencor/basic-yolo-keras/tree/master/experimental => YOLO 3

**Comment by muye5** [comment]
Expect the training part for v3....deeply.

**Comment by experiencor** [comment]
It's here https://github.com/experiencor/keras-yolo3
